cf/data/array/fullarray.py

```diff
@@ -0,0 +1,185 @@
+                dtype = source._get_component("dtype", None)
+            except AttributeError:
+                dtype = None
+
+            try:
+                shape = source._get_component("shape", None)
+            except AttributeError:
+                shape = None
+
+            try:
+                units = source._get_component("units", False)
+            except AttributeError:
+                units = False
+
+            try:
+                calendar = source._get_component("calendar", None)
+            except AttributeError:
+                calendar = None
+
+        self._set_component("fill_value", fill_value, copy=False)
+        self._set_component("dtype", dtype, copy=False)
+        self._set_component("shape", shape, copy=False)
+        self._set_component("units", units, copy=False)
+        self._set_component("calendar", calendar, copy=False)
+
+    def __getitem__(self, indices):
+        """x.__getitem__(indices) <==> x[indices]
+
+        Returns a numpy array.
+
+        """
+        # If 'apply_indices' is False then we can make a filled numpy
+        # array that already has the correct shape. Otherwise we need
+        # to make one with shape 'self.shape' and then apply the
+        # indices to that.
+        apply_indices = False
+
+        if indices is Ellipsis:
+            array_shape = self.shape
+        else:
+            array_shape = []
+            for i, size, i in zip(indices, self.shape):
+                if not isinstance(i, slice):
+                    continue
+
+                start, stop, step = i.indices(size)
+                a, b = divmod(stop - start, step)
+                if b:
+                    a += 1
+
+                array_shape.append(a)
+
+            if len(array_shape) != self.ndim:
+                apply_indices = True
+                array_shape = self.shape
+
+        fill_value = self.get_fill_value()
+        if fill_value is np.ma.masked:
+            array = np.ma.masked_all(array_shape, dtype=self.dtype)
+        elif fill_value is not None:
+            array = np.full(
+                array_shape, fill_value=fill_value, dtype=self.dtype
+            )
+        else:
+            array = np.empty(array_shape, dtype=self.dtype)
+
+        if apply_indices:
+            array = self.get_subspace(array, indices)
+
+        self._set_units()
+
+        return array
+
+    def __repr__(self):
+        """Called by the `repr` built-in function.
+
+        x.__repr__() <==> repr(x)
+
+        """
+        return f"<CF {self.__class__.__name__}{self.shape}: {self}>"
+
+    def __str__(self):
+        """Called by the `str` built-in function.
+
+        x.__str__() <==> str(x)
+
+        """
+        fill_value = self.get_fill_value()
+        if fill_value is None:
+            return "Uninitialised"
+
+        return f"Filled with {fill_value!r}"
+
+    def _set_units(self):
+        """TODO.
+
+        .. versionadded:: TODODASKVER
+
+        To cfdm.Array ?
```

Comment:
```suggestion
        TODODASKDOCS
```

unless you can remember where you were going with that thought :smile: